packages/bundler/contracts/tests/cross_sender_state/TestFakeWalletToken.sol

```diff
@@ -0,0 +1,48 @@
+// SPDX-License-Identifier: GPL-3.0
+pragma solidity ^0.8.12;
+
+import "@account-abstraction/contracts/interfaces/IAccount.sol";
+
+/// @dev A test contract that represents a potential attack where a wallet entity is also
+/// used as an associated storage by a different UserOperation.
+/// This allows this couple of UserOperations to escape the sandbox and invalidate a bundle.
+/// @notice this contract is copied as-is from the 'bundler-spec-tests' repo
+contract TestFakeWalletToken is IAccount {
+
+    mapping(address => uint256) private balances;
+    TestFakeWalletToken public anotherWallet;
+    address ep;
+
+    constructor(address _ep) payable {
+        ep = _ep;
+    }
+
+    function balanceOf(address _owner) public view returns (uint256 balance) {
+        return balances[_owner];
+    }
+
+    function sudoSetBalance(address _owner, uint256 balance) public {
+        balances[_owner] = balance;
+    }
+
+    function sudoSetAnotherWallet(TestFakeWalletToken _anotherWallet) public {
+        anotherWallet = _anotherWallet;
+    }
+
+    function validateUserOp(UserOperation calldata userOp, bytes32, uint256)
+    public override returns (uint256 validationData) {
+        if (userOp.callData.length == 20) {
```

Comment:
I think the calldata is not needed.
the code can be:
```
require(anotherWallet.balanceOf(this) > 0 )
balances[anotherWallet] = 0
```
and "anotherWallet" should be the owner of the 2nd userop to submit.

Reply:
This does not really work as both UserOps are marked as accessing "known sender storage", so you would get an empty bundle. I `map` the list of all "known senders" before bundling as was suggested in the issue description.